### DISCUSSION TITLE
Memory leak when using request.state

### DISCUSSION THREAD

**Discussion by galkahana** [encode/starlette#1748] [Potential Issue] [answered]
Hi, 
I noticed that setting properties on request.state results in memory leak.

Originally an issue I reported in [fastapi](https://github.com/tiangolo/fastapi/issues/4649#issuecomment-1177181535), it seems that the following server definition will cause memory usage to grow on every request:

```python
from starlette.applications import Starlette
from starlette.responses import JSONResponse
from starlette.routing import Route
from starlette.requests import Request


async def homepage(request: Request):
    request.state.test = [x for x in range(999999)]
    return JSONResponse({'hello': 'world'})


app = Starlette(routes=[
    Route('/', homepage),
])
```

The array set as the `test` property of `request.state` is never released. There's a workaround that I can use to resolve this by setting the the property to none at the end of the request - `request.state.test = None`.

I created a repo to demonstrate the issue [here](https://github.com/galkahana/starlette-state-leak).

Figured i'll report, so at least if someones getting a memory leak, this can help in possibly finding the cause.

**Comment by jhominal** [comment, marked as answer]
Oh, it seems that #1749 has been opened for this.
